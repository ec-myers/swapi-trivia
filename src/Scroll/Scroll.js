import React from 'react';
import PropTypes from 'prop-types';
import './Scroll.scss'


const Scroll = ({selectedMovie}) => {
  const {title, scrollText, episode} = selectedMovie;
  return (
    <div className="scroll">
      < div className="fade" ></div >
      <section className="star-wars">
        <div className="crawl">
          <div className="title">
            <p>Episode {episode}</p>
            <h1>{title}</h1>
          </div>
          <p>{scrollText}</p>
        </div>
      </section>
    </div>
  )
}

export default Scroll;

Scroll.propTypes = {
  selectedMovie: PropType.object
}