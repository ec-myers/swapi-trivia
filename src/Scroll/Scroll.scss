
.scroll {
  background: url(../Images/spaceImg.jpg);
  height: 100%;
  overflow: hidden;
  width: 100%;
}

.fade {
  position: relative;
  width: 100%;
  min-height: 60vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.star-wars {
  display: flex;
  justify-content: center;
  position: relative;
  height: 800px;
  color: #feda4a;
  font-size: 500%;
  font-weight: 600;
  line-height: 150%;
  perspective: 400px;
  text-align: justify;
}

.crawl {
  position: relative;
  top: 99999px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear infinite;
}

.crawl > .title {
  font-size: 90%;
  text-align: center;
}

.crawl > .title h1 {
  margin: 0 0 100px;
  text-transform: uppercase;
}

@keyframes crawl {
    0% {
    top: -100px;
    transform: rotateX(20deg)  translateZ(0);
  }
    100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}