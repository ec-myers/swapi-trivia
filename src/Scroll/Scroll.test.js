import React from 'react';
import { shallow } from 'enzyme';
import Scroll from './Scroll';

describe('Scroll', () => {

  let selectedMovie = {
    episode: 4,
    title: 'Revenge of the Sith',
    scrollText: 'Blah, Blah, Blah...'
  };

  let wrapper = shallow(
    <Scroll selectedMovie={selectedMovie} />
  );
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});