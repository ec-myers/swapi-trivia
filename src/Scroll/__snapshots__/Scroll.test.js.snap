// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll should match the snapshot 1`] = `
<div
  className="scroll"
>
  <div
    className="fade"
  />
  <section
    className="star-wars"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <p>
          Episode 
          4
        </p>
        <h1>
          Revenge of the Sith
        </h1>
      </div>
      <p>
        Blah, Blah, Blah...
      </p>
    </div>
  </section>
</div>
`;
